src/Foundation/NSUrlSessionHandler.cs

```diff
@@ -425,32 +424,39 @@ string GetHeaderSeparator (string name)
 			return value;
 		}
 
+		void AddManagedHeaders (NSMutableDictionary nativeHeaders, IEnumerable<KeyValuePair<string, IEnumerable<string>>> managedHeaders)
+		{
+			foreach (var keyValuePair in managedHeaders) {
+				var key = new NSString (keyValuePair.Key);
+				var value = new NSString (string.Join (GetHeaderSeparator (key), keyValuePair.Value));
+				nativeHeaders.Add (key, value);
```

Comment:
@mandel-macaque  Just a little nit, I think you can definitely avoid the NSString allocations with something like this

```suggestion
				var keyPtr = NSString.CreateNative (keyValuePair.Key);
				var valuePtr = NSString.CreateNative (string.Join (GetHeaderSeparator (keyValuePair.Key), keyValuePair.Value));
				nativeHeaders.LowlevelSetObject (valuePtr, keyPtr);
				NSString.ReleaseNative (keyPtr);
				NSString.ReleaseNative (valuePtr);
```

Reply:
I'll test the change before just in case ;)

Reply:
and confirmed :)